cmc-domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/offers/Settlement.java

```diff
@@ -33,11 +34,27 @@ public void reject(MadeBy party) {
     @JsonIgnore
     PartyStatement getLastStatement() {
         if (partyStatements.isEmpty()) {
-            throw new IllegalSettlementStatementException("No statements have yet been made during that settlement");
+            throw new IllegalSettlementStatementException(NO_STATEMENTS_MADE);
         }
         return partyStatements.get(partyStatements.size() - 1);
     }
 
+    @JsonIgnore
+    public PartyStatement getLastStatementWithOffer() {
+        if (partyStatements.isEmpty()) {
+            throw new IllegalSettlementStatementException(NO_STATEMENTS_MADE);
+        }
+
+        List<PartyStatement> tmpList = new ArrayList<>(partyStatements);
+        Collections.reverse(tmpList);
+
+        return tmpList.stream()
+            .filter((partyStatement -> partyStatement.getOffer().isPresent()))
```

Comment:
I know this is correct but I think I'd still filter on statement type, not offer being present. It's more strict, may be easier to detect some bugs in the future if any show up.

Reply:
done